jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2368,7 +2368,7 @@ public void mergeQueryParameters(String oldQuery, String newQuery, boolean updat
         if (newQueryParams == null || newQueryParams.size() == 0)
             mergedQueryParams = oldQueryParams == null ? NO_PARAMS : oldQueryParams;
         else if (oldQueryParams == null || oldQueryParams.size() == 0)
-            mergedQueryParams = newQueryParams == null ? NO_PARAMS : newQueryParams;
+            mergedQueryParams = newQueryParams;
```

Comment:
Previously, if there was no new queryString, the mergedQueryParams would be set to NO_PARAMS. Is this change intentional? Does the code handle the case where the mergedQueryParams are now null whereas before they were an empty MultiMap? Is there a test specifically for this?

Other than this, LGTM.

Reply:
That code change was because `newQueryParams` can never be null here because the if clause 4 lines early is `newQueryParams == null || `